rust/otel-arrow-rust/src/otlp/attributes/decoder.rs

```diff
@@ -62,3 +76,532 @@ where
+                AttributeValueType::Int => val_int_arr,
+                AttributeValueType::Bool => val_bool_arr,
+                AttributeValueType::Bytes => val_bytes_arr,
+                AttributeValueType::Double => val_double_arr,
+
+                // These types are always considered not equal for purposes of determining
+                // whether to delta encode parent ID
+                AttributeValueType::Map | AttributeValueType::Slice | AttributeValueType::Empty => {
+                    None
+                }
+            };
+
+            // add the first value from this range to the parent IDs
+            let mut curr_parent_id = parent_id_arr
+                .value_at(curr_range_start)
+                .expect("expect the batch not to be empty");
+            materialized_parent_ids.append_value(curr_parent_id);
+
+            if let Some(value_arr) = value_arr {
+                // if we have a value array here, we know the parent ID may be delta encoded
+                let range_length = idx + 1 - curr_range_start;
+                let values_range = value_arr.slice(curr_range_start, range_length);
+                let values_eq_next = create_next_element_equality_array(&values_range)?;
+
+                for batch_idx in (curr_range_start + 1)..(idx + 1) {
+                    let delta_or_parent_id = parent_id_arr.value_at_or_default(batch_idx);
+                    let prev_value_range_idx = batch_idx - 1 - curr_range_start;
+
+                    if values_eq_next.value(prev_value_range_idx)
+                        && !values_eq_next.is_null(prev_value_range_idx)
+                    {
+                        // value at current index equals previous so we're delta encoded
+                        curr_parent_id += delta_or_parent_id;
+                    } else {
+                        // value change (or null) breaks the sequence of delta encoding
+                        curr_parent_id = delta_or_parent_id;
+                    }
+                    materialized_parent_ids.append_value(curr_parent_id);
+                }
+            } else {
+                // if we're here, we've determined that the parent ID values are not delta encoded
+                // because the type doesn't support it
+                for batch_idx in (curr_range_start + 1)..(idx + 1) {
+                    materialized_parent_ids
+                        .append_value(parent_id_arr.value_at_or_default(batch_idx));
+                }
+            }
+
+            curr_range_start = idx + 1;
+        }
+    }
+    let materialized_parent_ids = Arc::new(materialized_parent_ids.finish());
+
+    // create new record batch but with parent column replaced
+    let parent_id_idx = record_batch
+        .schema()
+        .index_of(consts::PARENT_ID)
+        .expect("parent_id should be in the schema");
+    let columns = record_batch
+        .columns()
+        .iter()
+        .enumerate()
+        .map(move |(i, col)| {
```

Comment:
I don't think we require `move` here.
```suggestion
        .map(|(i, col)| {
```